contracts/Account.sol

```diff
@@ -743,6 +862,83 @@ contract Account is UUPSOwnableUpgradeable, UsingRegistryUpgradeable, Managed, I
             uint256
         )
     {
-        return (1, 1, 2, 0);
+        return (1, 2, 0, 0);
+    }
+
+    /**
+     * @notice Returns the toVote amount of CELO directed towards `group`. This is
+     * the toVote CELO balance for `group` minus the toRevoke amount. ToRevoke is updated.
+     * @param group The address of the validator group.
+     * @return The toVote amount of CELO directed towards `group`.
+     */
+    function getToVote(address group) private returns (uint256) {
+        uint256 finalToVote = scheduledVotes[group].toVote;
+
+        if (scheduledVotes[group].toRevoke > 0) {
+            uint256 toSubtract = Math.min(finalToVote, scheduledVotes[group].toRevoke);
+            scheduledVotes[group].toRevoke -= toSubtract;
+            finalToVote -= toSubtract;
+            scheduledVotes[group].toVote = finalToVote;
+        }
+
+        return finalToVote;
+    }
+
+    /**
+     * @notice Returns the toRevoke amount of CELO directed towards `group`. This is
+     * the toRevoke CELO balance for `group` minus the toVote amount. ToVote is updated.
+     * @param group The address of the validator group.
+     * @return The toRevote amount of CELO directed towards `group`.
+     */
+    function getToRevoke(address group) private returns (uint256) {
+        uint256 finalToRevoke = scheduledVotes[group].toRevoke;
+
+        if (scheduledVotes[group].toVote > 0) {
+            uint256 toSubtract = Math.min(finalToRevoke, scheduledVotes[group].toVote);
+            scheduledVotes[group].toVote -= toSubtract;
+            finalToRevoke -= toSubtract;
+            scheduledVotes[group].toRevoke = finalToRevoke;
+        }
+
+        return finalToRevoke;
+    }
+
+    /**
+     * @notice Adds amount to toVote and returns the toVote amount of CELO directed towards `group`.
+     * This is the toVote CELO balance for `group` minus the toRevoke amount. ToRevoke is updated.
+     * @param toAdd The amount to add to toVote.
+     * @param group The address of the validator group.
+     * @return The toVote amount of CELO directed towards `group`.
+     */
+    function addToVote(address group, uint256 toAdd) private returns (uint256) {
+        uint256 finalToVote = scheduledVotes[group].toVote + toAdd;
+
+        if (scheduledVotes[group].toRevoke > 0) {
+            uint256 toSubtract = Math.min(finalToVote, scheduledVotes[group].toRevoke);
```

Comment:
There's several functions here that all do similar logic of updating the toVote/Revoke amounts. Could these be cleaned up with an `updateToVoteAndRevoke` helper function?

Reply:
I can create one but usually only one of the functions is used during the transaction. It means it will cost a bit more gas. Do you think it is worth it ?

Reply:
I combined it into getAndUpdateToVoteAndToRevoke